habitat_sim/registry.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import collections
+import re
+from typing import Optional, Type
+
+from habitat_sim.agent.controls import SceneNodeControl
+
+__all__ = ["registry"]
+
+
+def _camel_to_snake(name):
+    s1 = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", name)
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", s1).lower()
+
+
+class registry:
+    r"""registry is a central source of truth in Habitat-Sim
+
+    Taken from Pythia, it is inspired from Redux's
+    concept of global store. registry maintains mappings of various information
```

Comment:
> This registry is slightly different than the one in habitat-api

AH, that's why this docstring felt like a deja vu to me :)

Reply:
Yep :)